packagehandlers/pnpmpackagehandler.go

```diff
@@ -0,0 +1,108 @@
+package packagehandlers
+
+import (
+	"errors"
+	"fmt"
+	"github.com/jfrog/frogbot/v2/utils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	pnpmDependencyPattern    = "\\s*\"%s\"\\s*:\\s*\"[~|^]?%s\""
+	pnpmDescriptorFileSuffix = "package.json"
+	nodeModulesPathPattern   = ".*node_modules.*"
+)
+
+type PnpmPackageHandler struct {
+	CommonPackageHandler
+}
+
+func (pnpm *PnpmPackageHandler) UpdateDependency(vulnDetails *utils.VulnerabilityDetails) error {
+	if vulnDetails.IsDirectDependency {
+		return pnpm.updateDirectDependency(vulnDetails)
+	}
+
+	return &utils.ErrUnsupportedFix{
+		PackageName:  vulnDetails.ImpactedDependencyName,
+		FixedVersion: vulnDetails.SuggestedFixedVersion,
+		ErrorType:    utils.IndirectDependencyFixNotSupported,
+	}
+}
+
+func (pnpm *PnpmPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails) (err error) {
```

Comment:
I know that we've done it all the time, but let's avoid implicit returns (using "return" without specifying values) whenever possible. It's a bad practice that we preferably shouldn't continue.